docs/rfcs/20220208_admin_api_basic_auth.md

```diff
@@ -0,0 +1,234 @@
+
+This is not possible using existing stored SCRAM credentials, because
+HTTP digest hashing is different from the hashing used in SCRAM credentials.
+
+We could implement this for newly created credentials if we calculated
+and stored the HTTP Digest MD5 hash of the password during account creation
+(at which time, we have the plaintext password).
+
+While HTTP digest auth is slightly better than basic auth in that it doesn't
+transmit passwords in the clear, it has many security issues of its own, and
+in practice either of these protocols should be transported over TLS for
+a meaningful level of security.
+
+### Alternative: restrict admin API to unauthenticated localhost listener
+
+If the admin API only listened on 127.0.0.1, then users could SSH to a redpanda
+node and drive admin operations from there.  This has several downsides:
+- Security is still weak, as any process running on the redpanda node can
+  access the admin API
+- It is inconvenient to do all operations via an SSH session
+- The admin API can't issue a redirect to another node (e.g. to redirect
+  to controller leader) if all the nodes are only listening on localhost.
+
+### Alternative: create separate admin API credentials (e.g. API keys)
+
+It is convenient but not essential to use the same system of credentials
+for the admin API that we use for Kafka protocol.  We could implement
+a separate pool of API keys.
+
+This approach would increase the surface area of the feature substantially,
+as we would not be able to use the existing user CRUD commands, and would
+be more complex to explain to the user compared with the existing user/password
+framework they know from the world of Kafka.
+
+### Alternative: shared secret in node config
+
+For use cases like the kubernetes operator, it might be sufficient to simply
+store a random string in the node configuration file (redpanda.yaml), and
+use this as a shared secret.
+
+This has several downsides:
+* We would be storing a de-facto cleartext password
+* Key changes would require node restart
+* User would be unable to distinguish clients (for audit purposes) by
+  user account if all shared a statically set token for admin API access.
+
+### Alternative: only authenticate writes
+
+For some use cases (e.g. monitoring), it might be convenient to give the user read-only
+access without providing a secret.
+
+However, this cannot be applied uniformly to all endpoints, for example the config
+endpoint includes sensitive configuration properties like the cloud access key.  It would
+also remove the ability to record an audit record of which users were accessing what
+information.
+
+### Alternative: per-endpoint RBAC
+
+This isn't so much an alternative as a possible extension.  The authorization layer
+can be extended in many ways in the future: this RFC is just about getting some
+kind of authentication (SASL user account) combined with the most basic level
+of authorization (the superuser list).
+
```

Comment:
Eventually we will want to build an RBAC system on top of our auth platform. Agreed that we don't need it yet, and building the basic auth layer to be extensible is the way to go in preserving long-term flexibility.